tensorflow_addons/image/distort_image_ops.py

```diff
@@ -99,6 +102,43 @@ def random_hsv_in_yiq(
         )
 
 
+def _adjust_hsv_in_yiq(
+    image,
+    delta_hue,
+    scale_saturation,
+    scale_value,
+):
+    if image.shape.rank is not None and image.shape.rank < 3:
+        raise ValueError("input must be at least 3-D.")
+    if image.shape[-1] is not None and image.shape[-1] != 3:
+        raise ValueError(
+            "input must have 3 channels but instead has {}.".format(image.shape[-1])
+        )
+    # Construct hsv linear transformation matrix in YIQ space.
+    # https://beesbuzz.biz/code/hsv_color_transforms.php
+    yiq = tf.constant(
```

Comment:
Have you tested if it will be slower using https://www.tensorflow.org/api_docs/python/tf/image/rgb_to_yiq ?

Reply:
Some slowdown. I observe there are some redundant ops being used in `tf.image.rgb_to_yiq` and `tf.image.yiq_to_rgb`.

Reply:
Ok this is similar to https://github.com/tensorflow/tensorflow/blob/v2.4.1/tensorflow/python/ops/image_ops_impl.py#L2465-L2529
But as I see that has only native ops..